bindings/rust/s2n-tls/src/connection.rs

```diff
@@ -926,6 +926,14 @@ impl Connection {
         })
     }
 
+    pub fn selected_application_protocol(&self) -> Result<Option<&str>, Error> {
```

Comment:
In this case, I don't think it makes sense to return a `Result`, as there aren't any fail-able operations. We generally return `Result`s because most of our API's return error codes, but in this case we directly return `const *` without extra information indicating success or failure.

I also don't think we want to return a `&str`. In rust a valid `str` is UTF-8 encoded, but I don't think we have that guarantee here. 

All in all, I think this should be returning `Option<&[u8]>` rather than the `Result<...>` type.